src/greenlight/step.clj

```diff
@@ -268,15 +268,20 @@
 
 (defn initialize
   "Resolves contextual properties of a step prior to execution."
-  [step ctx]
-  (cond-> step
-    (fn? (::title step)) (update ::title #(% ctx))))
+  [system step ctx]
+  (let [inputs (collect-inputs system ctx step)]
+    [(cond-> step
+       (fn? (::title step)) (update ::title #(try
+                                               (% ctx)
```

Comment:
The step `::inputs` are just merged into the context from a step's perspective, so it feels a little off here to have a two arity function that accepts both for the title. It also introduces this `try` / `catch` on the `ArityException`. 

What do you think of resolving the step inputs in initialization instead, merging those into the context and allowing a function of that as the title? I put this together in c000155.

Reply:
I do find that implementation much simpler and preferred. I will say, however, I was initially hesitant to merge the context and the inputs as I don't actually see that happening anywhere currently. `collect-inputs` resolves inputs based on the context, but it appears the map passed to the test function contains only the inputs.

Do you want me to put together a new PR with that commit and and small update to the one test contextualizing the title to test that inputs are available, too?

Reply:
> I was initially hesitant to merge the context and the inputs as I don't actually see that happening anywhere currently

Right, previously we allowed the title to be composed of any entry from the context, but not input overrides passed in from `defstep` -- I think that's a little funny, because it means you can pull from the context pieces of context that aren't exposed to test functions, without being able to pull the explicitly passed inputs. 

Extra inputs from `defstep` are merged in here: https://github.com/amperity/greenlight/blob/develop/src/greenlight/step.clj#L114-L116

The change in c000155 would continue allowing the test title function to pull arbitrary keys from the context, it just makes sure the overrides are applied before that happens.

You can update this PR if none of that sounds crazy - cherrypick or pull the changes, I think the test you've added is valuable.